src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandleArray.cs

```diff
@@ -146,5 +146,39 @@ public static partial class JsonSerializer
 
             return false;
         }
+
+        // If this method is changed, also change JsonPropertyInfoNullable.ApplyValue and JsonPropertyInfoNotNullable.ApplyValue
+        internal static void ApplyObjectToEnumerable(object value, JsonSerializerOptions options, ref ReadStackFrame frame, bool setPropertyDirectly = false)
+        {
+            if (frame.IsEnumerable())
+            {
+                if (frame.TempEnumerableValues != null)
+                {
+                    frame.TempEnumerableValues.Add(value);
```

Comment:
I see casts to `IList/IList<T>` elsewhere in similar looking code. Why can we avoid the cast here?

Reply:
`TempEnumerableValues` is declared as `IList` which takes `object` for its enumerator and `value` here is `object`. This method is called when adding a list or object so there is not a concern about boxing like the other similar locations.